packages/runtimes/webextension/src/WebExtensionRuntime.js

```diff
@@ -12,39 +12,50 @@ const AUTORELOAD_BG = fs.readFileSync(
 
 export default (new Runtime({
   apply({bundle, bundleGraph, options}) {
-    if (
-      bundle.type != 'js' ||
-      !options.hmrOptions ||
-      !bundle.env.isBrowser() ||
-      bundle.env.isWorklet()
-    ) {
+    if (!bundle.env.isBrowser() || bundle.env.isWorklet()) {
       return;
     }
-    const manifest = bundleGraph
-      .getBundles()
-      .find(b => b.getMainEntry()?.meta.webextEntry === true);
-    const entry = manifest?.getMainEntry();
-    const insertDep = entry?.meta.webextBGInsert;
-    if (insertDep == null) return;
-    const insertBundle = bundleGraph.getReferencedBundle(
-      nullthrows(entry?.getDependencies().find(dep => dep.id === insertDep)),
-      nullthrows(manifest),
-    );
-    let firstInsertableBundle;
-    bundleGraph.traverseBundles((b, _, actions) => {
-      if (b.type == 'js') {
-        firstInsertableBundle = b;
-        actions.stop();
-      }
-    }, insertBundle);
+    if (bundle.name == 'manifest.json') {
+      const asset = bundle.getMainEntry();
+      if (asset?.meta.webextEntry !== true) return;
 
-    // Add autoreload
-    if (bundle === firstInsertableBundle) {
+      // Hack to bust packager cache when any descendants update
+      const descendants = [];
+      bundleGraph.traverseBundles(b => {
+        descendants.push(b.id);
+      }, bundle);
       return {
         filePath: __filename,
-        code: AUTORELOAD_BG,
+        code: JSON.stringify(descendants),
         isEntry: true,
       };
```

Comment:
Nice hack. This is ok for now. We'll think about a better way to do this in core.